src/v/kafka/server/group.cc

```diff
@@ -2149,6 +2150,32 @@ group::handle_commit_tx(cluster::commit_group_tx_request r) {
     }
 }
 
+error_code
+group::check_group_before_commit_offset(const txn_offset_commit_request& r) {
+    kafka::member_id mid(r.data.member_id);
+    if (r.data.group_instance_id.has_value()) {
+        auto static_member_it = _static_members.find(
+          kafka::group_instance_id(r.data.group_instance_id.value()));
```

Comment:
you may add mapping for the `group_instance_id` to be interpreted as correct type in RPC code generator.